pkg/onepassword/client/connect/connect.go

```diff
@@ -0,0 +1,85 @@
+package connect
+
+import (
+	"fmt"
+
+	"github.com/1Password/connect-sdk-go/connect"
+	"github.com/1Password/connect-sdk-go/onepassword"
+	"github.com/1Password/onepassword-operator/pkg/onepassword/model"
+)
+
+// Config holds the configuration for the Connect client.
+type Config struct {
+	ConnectHost  string
+	ConnectToken string
+	UserAgent    string
+}
+
+// Connect is a client for interacting with 1Password using the Connect API.
+type Connect struct {
+	client connect.Client
+}
+
+// NewClient creates a new Connect client using provided configuration.
+func NewClient(config Config) *Connect {
+	return &Connect{
+		client: connect.NewClient(config.ConnectHost, config.ConnectToken),
+	}
+}
+
+func (c *Connect) GetItemByID(vaultID, itemID string) (*model.Item, error) {
+	connectItem, err := c.client.GetItemByUUID(itemID, vaultID)
+	if err != nil {
+		return nil, fmt.Errorf("1password Connect error: %w", err)
+	}
+
+	var item model.Item
+	item.FromConnectItem(connectItem)
+	return &item, nil
+}
+
+func (c *Connect) GetItemsByTitle(vaultID, itemTitle string) ([]model.Item, error) {
+	// Get all items in the vault with the specified title
+	connectItems, err := c.client.GetItemsByTitle(itemTitle, vaultID)
+	if err != nil {
+		return nil, fmt.Errorf("1password Connect error: %w", err)
+	}
+
+	var items []model.Item
+	for _, connectItem := range connectItems {
+		var item model.Item
+		item.FromConnectItem(&connectItem)
+		items = append(items, item)
+	}
```

Comment:
`[nit]`

I think this might be improved a bit:
```suggestion
	items := make([]model.Item, len(connectItems))
	for i, connectItem := range connectItems {
		var item model.Item
		item.FromConnectItem(&connectItem)
		items[i] = item
	}
```

And would it make more sense to have the function `FromConnectItem` return the `model.Item` instead of doing in-memory replacement? 🤔

Reply:
I'd say this:
```
var item model.Item
item.FromConnectItem(&connectItem)
```

is more common Go implementation, and I like that `model.Item` have specific methods that allows to create `model.Item` from different models (connect and sdk) in our case. So, I'll leave it as it is.